.github/workflows/test.yml

```diff
@@ -107,25 +116,70 @@ jobs:
           reporter: github-pr-check
 
       - name: Install grcov
-        if: matrix.rust == 'stable' && matrix.test == 'std' && runner.os == 'Linux'
+        if: runner.os == 'Linux' && matrix.rust == 'stable' && matrix.test == 'std'
         env:
           GRCOV_LINK: https://github.com/mozilla/grcov/releases/download
           GRCOV_VERSION: v0.8.18
         run: |
           curl -L "$GRCOV_LINK/$GRCOV_VERSION/grcov-x86_64-unknown-linux-musl.tar.bz2" |
           tar xj -C $HOME/.cargo/bin
 
+      - name: (windows) install warp
+        if: runner.os == 'Windows'
+        shell: bash
+        run: |
+          set -e
+
+          curl.exe -L https://www.nuget.org/api/v2/package/Microsoft.Direct3D.WARP/1.0.7.1 -o warp.zip
+          7z.exe e warp.zip -owarp build/native/amd64/d3d10warp.dll
+
+          mkdir -p target/debug/deps
+
+          cp -v warp/d3d10warp.dll target/debug/
+          cp -v warp/d3d10warp.dll target/debug/deps
+
+      - name: (windows) install mesa
+        if: runner.os == 'Windows'
+        shell: bash
+        run: |
+          set -e
+
+          curl.exe -L https://github.com/pal1000/mesa-dist-win/releases/download/23.2.1/mesa3d-23.2.1-release-msvc.7z -o mesa.7z
+          7z.exe e mesa.7z -omesa x64/{opengl32.dll,libgallium_wgl.dll,libglapi.dll,vulkan_lvp.dll,lvp_icd.x86_64.json}
+
+          mkdir -p target/debug/deps
+
+          cp -v mesa/* target/debug/
+          cp -v mesa/* target/debug/deps
+
+          echo "VK_DRIVER_FILES=$PWD/mesa/lvp_icd.x86_64.json" >> "$GITHUB_ENV"
+          echo "GALLIUM_DRIVER=llvmpipe" >> "$GITHUB_ENV"
+
+      - name: (windows) install directx
+        if: runner.os == 'Windows'
+        uses: napokue/setup-dxc@v1.1.0
+
+      - name: (windows) run wgpu-info
+        if: runner.os == 'Windows'
+        shell: bash
+        run: |
+          echo "$PATH"
+          cargo install wgpu-info
+          which wgpu-info
+          export RUST_LOG=trace
+          wgpu-info
```

Comment:
Do we want to leave this step as check or can we remove it?

Reply:
I'll remove it, it was more for debugging purpose.

Reply:
Removed.